src/buf/into_buf.rs

```diff
@@ -1,6 +1,14 @@
 use super::{Buf};
 
-use std::io;
+#[cfg(feature = "use_alloc")]
+use alloc::string::String;
+#[cfg(feature = "use_alloc")]
+use alloc::vec::Vec;
+
+#[cfg(not(feature = "use_std"))]
+use buf::Cursor;
```

Comment:
A couple options here:

1) Gate this (and anything that depends on it) on `use_alloc`
2) Stop using `std::io::Cursor` entirely (i.e. fix #75)